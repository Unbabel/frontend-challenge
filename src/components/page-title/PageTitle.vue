<script setup>
defineProps({
  text: {
    type: String,
    required: true,
    default: "",
  },
});
</script>

<template>
  <span class="page-title" data-test="page-title">
    {{ text }}
  </span>
</template>

<style scoped>
.page-title {
  font-size: 1.125rem;
  font-weight: 500;
  color: var(--palette-dark);
}
</style>
