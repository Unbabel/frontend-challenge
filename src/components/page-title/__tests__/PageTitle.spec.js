import { describe, it, expect, beforeEach } from "vitest";

import { mount } from "@vue/test-utils";

// Components
import PageTitle from "../PageTitle.vue";

describe("PageTitle", () => {
	let wrapper;
	let testProps = { text: "My Title" };

	beforeEach(() => {
		// Arrange
		wrapper = mount(PageTitle, {
			propsData: testProps,
		});
	});

	it("matches snapshot", () => {
		// Assert
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("renders properly", () => {
		// Assert
		expect(wrapper.text()).toContain("My Title");
	});

	it("should render empty string when no text is passed", () => {
		// Arrange
		wrapper = mount(PageTitle, {
			propsData: {},
		});

		// Assert
		const pageTitleStr = wrapper.find('[data-test="page-title"]');
		expect(pageTitleStr.text()).toBe("");
	});

	it("has correct classes", () => {
		// Assert
		expect(wrapper.classes()).toContain("page-title");
	});
});
