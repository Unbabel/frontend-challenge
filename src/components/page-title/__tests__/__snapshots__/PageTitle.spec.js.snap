// Vitest Snapshot v1

exports[`PageTitle > matches snapshot 1`] = `"<span data-v-dde5e378=\\"\\" data-test=\\"page-title\\" class=\\"page-title\\"> My Title </span>"`;
